app/api/services/recipesService.js

```diff
@@ -1,5 +1,20 @@
 import { Recipe } from '../models/recipes.js';
 import { User } from '../models/users.js';
+import { sequelize } from '../db/db.js';
+import { QueryTypes } from 'sequelize';
+
+async function listRecipes(limit, offset, whereCondition = null) {
+    return await sequelize.query(
+        `SELECT "recipes".*, 
+                "user"."id" AS "user_id", 
+                "user"."name" AS "user_name", 
+                "user"."avatar" AS "user_avatar", 
+                "user"."email" AS "user_email" FROM "recipes" AS "recipes" LEFT OUTER JOIN "users" AS "user" ON "recipes"."owner" = "user"."id" ${
+                    whereCondition ? `WHERE ${whereCondition}` : ''
+                } LIMIT ${limit} OFFSET ${offset}`,
+        { type: QueryTypes.SELECT }
+    );
+}
```

Comment:
Різні функції сервісу будуть видавати дані в різних форматах. Це не дуже ОК

```suggestion
    const response = await sequelize.query(
        `SELECT "recipes".*, 
                "user"."id" AS "user_id", 
                "user"."name" AS "user_name", 
                "user"."avatar" AS "user_avatar", 
                "user"."email" AS "user_email" FROM "recipes" AS "recipes" LEFT OUTER JOIN "users" AS "user" ON "recipes"."owner" = "user"."id" ${
                    whereCondition ? `WHERE ${whereCondition}` : ''
                } LIMIT ${limit} OFFSET ${offset}`,
        { type: QueryTypes.SELECT }
    );
    if (!response || response.length === 0) {
        return [];
    }
    return response.map(recipe => ({
        id: recipe.id,
        owner: {
            id: recipe.user_id,
            name: recipe.user_name,
            avatar: recipe.user_avatar,
        },
        title: recipe.title,
        description: recipe.description,
        thumb: recipe.thumb,
    }));
}
```